inlong-tubemq/tubemq-client-twins/tubemq-client-go/client/consumer_impl.go

```diff
@@ -280,7 +280,8 @@ func (c *consumer) Confirm(confirmContext string, consumed bool) (*ConsumerResul
 		return nil, errs.New(errs.RetErrConfirmTimeout, "Not found the partition by confirm_context!")
 	}
 
-	rsp, err := c.sendConfirmReq2Broker(partition)
+	defer c.rmtDataCache.ReleasePartition(true, c.subInfo.IsFiltered(topic), confirmContext, consumed)
```

Comment:
Sending a confirm request after release is prone to misconfirmation problems:
The partition is released, the second getMessage starts, the first confirm arrives, and the second getMessage is confirmed

Reply:
Partition will be released before leaving this function, so the request will be sent to broker.

Reply:
ye, not a problem. I checked the defer usage of Go, which is similar to the finally approach in Java, it will be called before processing exit.